### PR TITLE
Initial commit submodule

### PR DESCRIPTION
**Pull request by ldebello-ddl**
As part of Domino Cloud we want to automate the way we offer nexus, currently we are using Private Link to connect Control Plane and Data Plane in a secure way.

Private Link Requirements:

* Use NLB
* Use VPC Endpoint Services

Notes:

We disabled two TF validations because we want the behaviour defined in TF 

https://docs.bridgecrew.io/docs/bc_aws_networking_62
We want to allow TF to destroy NLB 

https://docs.bridgecrew.io/docs/ensure-that-vpc-endpoint-service-is-configured-for-manual-acceptance
We want VPC to automatically get connections without manual approvals

PR Feedback from: https://github.com/dominodatalab/terraform-aws-eks/pull/110

### CODE REVIEW

File: submodules/privatelink/iam.tf [hunk 1]
```diff
@@ -0,0 +1,50 @@
+data "aws_iam_policy_document" "load_balancer_controller" {
+  statement {
+    sid     = ""
+    effect  = "Allow"
+    actions = ["sts:AssumeRoleWithWebIdentity"]
+
+    condition {
+      test     = "StringEquals"
+      variable = "${replace(var.oidc_provider_id, "/arn:.*:oidc-provider//", "")}:sub"
```
Comment (steved) [inline submodules/privatelink/iam.tf:9]: could store `replace(var.oidc_provider_id, "/arn:.*:oidc-provider//", "")` as a local

File: submodules/privatelink/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,6 @@
+output "info" {
+  description = "Target groups..."
```
Comment (steved) [inline submodules/privatelink/outputs.tf:2]: if you can, please fill this in before merging

File: submodules/privatelink/vpc-endpoint-services.tf [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+  tags = {
+    "Name" = "${var.deploy_id}-${each.key}"
+  }
+}
+
+resource "aws_route53_record" "service_endpoint_private_dns_verification" {
+  for_each = local.endpoint_services
+
+  zone_id = data.aws_route53_zone.hosted[0].zone_id
```
Comment (steved) [inline submodules/privatelink/vpc-endpoint-services.tf:146]: I think this assumes `route53_hosted_zone_name` is non-null (which is fine, but maybe just make the variable required?)

File: submodules/privatelink/vpc-endpoint-services.tf [hunk 4]
```diff
@@ -0,0 +1,153 @@
+locals {
+  endpoint_services = tomap({ for service in var.vpc_endpoint_services : service.name => service.private_dns })
```
Comment (steved) [inline submodules/privatelink/vpc-endpoint-services.tf:2]: is this `tomap` needed?

File: submodules/privatelink/vpc-endpoint-services.tf [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+    }
+
+    principals {
+      type        = "Service"
+      identifiers = ["delivery.logs.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_s3_bucket" "lb_logs" {
```
Comment (steved) [inline submodules/privatelink/vpc-endpoint-services.tf:77]: Can we reuse our existing monitoring bucket instead? Just keeps things in one place for now and infosec can just pull from it

**Review by steved — COMMENTED** [review]
couple small comments

File: submodules/privatelink/outputs.tf (hunk 2 above)
Comment (ldebello-ddl) [reply submodules/privatelink/outputs.tf:2]: it only return target groups, I added "..." based on other outputs that use the same pattern, I can remove it if you refer to that

File: submodules/privatelink/iam.tf (hunk 1 above)
Comment (ldebello-ddl) [reply submodules/privatelink/iam.tf:9]: fixed

File: submodules/privatelink/vpc-endpoint-services.tf (hunk 3 above)
Comment (ldebello-ddl) [reply submodules/privatelink/vpc-endpoint-services.tf:146]: fixed

File: submodules/privatelink/vpc-endpoint-services.tf (hunk 4 above)
Comment (ldebello-ddl) [reply submodules/privatelink/vpc-endpoint-services.tf:2]: making it a map it is easier to work in the tf

Comment (ldebello-ddl) [reply submodules/privatelink/vpc-endpoint-services.tf:2]: any suggestion¿

File: submodules/privatelink/vpc-endpoint-services.tf (hunk 5 above)
Comment (ldebello-ddl) [reply submodules/privatelink/vpc-endpoint-services.tf:77]: yes probably I will replace it

File: submodules/privatelink/vpc-endpoint-services.tf (hunk 4 above)
Comment (steved) [reply submodules/privatelink/vpc-endpoint-services.tf:2]: Oh I just wasn't sure why the tomap was needed I thought the map was created with
```
endpoint_services = { for service in var.vpc_endpoint_services : service.name => service.private_dns }
```

Comment (ldebello-ddl) [reply submodules/privatelink/vpc-endpoint-services.tf:2]: let me check

**Pull request closed without merging**